geotrek/trekking/parsers.py

```diff
@@ -690,13 +690,16 @@ def filter_geom(self, src, val):
         geom_file = self._fetch_geometry_file(plan)
 
         ext = self._get_plan_extension(plan)
-        if ext == 'gpx':
-            return ApidaeTrekParser._get_geom_from_gpx(geom_file)
-        elif ext == 'kml':
-            return ApidaeTrekParser._get_geom_from_kml(geom_file)
-        elif ext == 'kmz':
-            kml_file = zipfile.ZipFile(io.BytesIO(geom_file)).read('doc.kml')
-            return ApidaeTrekParser._get_geom_from_kml(kml_file)
+        try:
+            if ext == 'gpx':
+                return get_geom_from_gpx(geom_file)
+            elif ext == 'kml':
+                return get_geom_from_kml(geom_file)
+            elif ext == 'kmz':
+                kml_file = zipfile.ZipFile(io.BytesIO(geom_file)).read('doc.kml')
+                return get_geom_from_kml(kml_file)
+        except ValueError as e:
```

Comment:
The messages of errors raised by the get geom functions are user-readable and intended to be displayed in the parser import output. So raising the error again as a `RowImportError` with the original message is good. :+1: 
IMO `ValueError` is too broad for this usecase. The `try` clause is above too much code to use it. The risk is to catch `ValueError` exceptions from unexpected part of the code and to show them as is to the user which is not great. And developers will struggle to find the origin of such errors because they won't have a traceback in the import output.

I suggest to add a new type

```python
class GeomValueError(Exception):
  pass
```

in the `utils` module and to `try..catch` that instead.

Reply:
Good point, thank you !